include/queue_fn.php

```diff
@@ -24,26 +24,18 @@ function remove_queue_item($id) {
  */
 function was_recently_delayed($cid) {
 
-	$was_delayed = false;
-
-	// Are there queue entries that were recently added?
 	$r = q("SELECT `id` FROM `queue` WHERE `cid` = %d
-		AND `last` > UTC_TIMESTAMP() - interval 15 minute LIMIT 1",
+		AND `last` > UTC_TIMESTAMP() - INTVAL 15 MINUTE LIMIT 1",
 		intval($cid)
 	);
+	if (dbm::is_result($r))
+		return true;
 
-	$was_delayed = dbm::is_result($r);
-
-	// We set "term-date" to a current date if the communication has problems.
-	// If the communication works again we reset this value.
-	if ($was_delayed) {
-		$r = q("SELECT `term-date` FROM `contact` WHERE `id` = %d AND `term-date` <= '1000-01-01' LIMIT 1",
-			intval($cid)
-		);
-		$was_delayed = !dbm::is_result($r);
-	}
+	$r = q("SELECT `term-date` FROM `contact` WHERE `id` = %d AND `term-date` != '' AND `term-date` != '0000-00-00 00:00:00' LIMIT 1",
+		intval($cid)
+	);
```

Comment:
Could you implement the function as it was originally? Originally it works the following way:

At first it sets "$was_delayed" when there was a queue entry earlier than 15 minutes ago. But when "term-date" isn't set to a current date we don't care about it. This has the background that we set "term-date" to the current timestamp whenever there is a communication problem - but we reset it to "0000-00-00" when it works again.

When we would implement it the way you did it, we never would deliver any content at all in the queue when "term-date" was set once.

And BTW: Since current versions of MySQL can't handle the "0000-00-00" timestamp value we slowly change it to "1000-01-01".